### PR TITLE
Progression capabilities

### PR DESCRIPTION
**Pull request by RaulPerez1**
**Customer scenario**

.Net core projects don't implement IntelliSense type so progression features don't work in those (i.e. seeing the types in a file in solution explorer)

**Bugs this fixes:** 

https://github.com/dotnet/roslyn-project-system/issues/592

**Workarounds, if any**

No workarounds

**Risk**

Low. No actual changes to the code here other than move the provider to use capabilities instead of IntelliSense type on the MEF export.

**Performance impact**

Low - See above for details.

**Is this a regression from a previous update?**

Not a regression.

**Root cause analysis:**

Not a bug in Roslyn per-se. The change to support capabilities was added to progression in Dev15.

**How was the bug found?**

Dogfooding

### CODE REVIEW

**Comment by RaulPerez1** [comment]
@Pilchie, @jaredpar, @jmarolf 

This is the same change as https://github.com/dotnet/roslyn/pull/15935 - That one got automatically closed as I was trying to recreate the branch with the changes post Jared's move to Dev15.

File: src/VisualStudio/Core/Next/Progression/RoslynGraphProvider.cs [hunk 1]
```diff
@@ -12,11 +11,11 @@
 
 namespace Microsoft.VisualStudio.LanguageServices.CSharp.Progression
 {
-    [GraphProvider(Name = "CSharpRoslynProvider", IntellisenseType = Guids.CSharpProjectRootIdString)]
-    internal sealed class CSharpGraphProvider : AbstractGraphProvider
+    [GraphProvider(Name = "RoslynGraphProvider", ProjectCapability = "CSharp | VisualBasic")]
```
Comment (jmarolf) [inline src/VisualStudio/Core/Next/Progression/RoslynGraphProvider.cs]: >GraphProvider [](start = 5, length = 13)

Why don't we need IntellisenseType for legacy project systems?

Comment (RaulPerez1) [inline src/VisualStudio/Core/Next/Progression/RoslynGraphProvider.cs]: The legacy project system also has the capability so a capability check works for both the new and the old project systems.

Comment (RaulPerez1) [inline src/VisualStudio/Core/Next/Progression/RoslynGraphProvider.cs]: The legacy project system also has the capability so a capability check works for both the new and the old project systems.

**Comment by RaulPerez1** [comment]
@dotnet/roslyn-ide The first 2 builds fail with this which is lacking an actual error. How shoud I proceed?

16:59:57   Toolset -> D:\j\workspace\windows_deter---bd1d49d3\Binaries\Debug\Exes\Toolset\Toolset_DoNetUse.exe
17:00:00 D:\j\workspace\windows_deter---bd1d49d3\Binaries\Debug\Exes\Toolset\csc.exe
...
17:00:00        57 file(s) moved.
17:00:02 Killing all build-related processes
17:00:02 ERROR: The process "MSBuild.exe" with PID 3412 could not be terminated.
17:00:02 Reason: The operation attempted is not supported.

17:00:02 Build failed with ERRORLEVEL 1
17:00:02 Killing all build-related processes
17:00:02 ERROR: The process "msbuild.exe" not found.
17:00:02 Build step 'Execute Windows batch command' marked build as failure

**Comment by jasonmalinowski** [comment]
That looks like a flaky issue...maybe there's a race in how it kills stuff...?

@dotnet-bot retest windows_determinism_prtest please.

**Comment by RaulPerez1** [comment]
From the offline conversation, reverted the changes to move the class to the "Next" assembly and made them in place now that the original projects are targeting 15.0 assemblies.

**Comment by RaulPerez1** [comment]
Tagging @MattGertz @srivatsn for shiproom

**Comment by MattGertz** [comment]
General failure of a much-used feature in this case, so we'll bring up.

**Comment by natidea** [comment]
Approved to merge via [371300](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/371300)

**Pull request merged**